resources/type-system/inference.md

```diff
@@ -474,42 +462,78 @@ replaced with `Null`, and every covariant occurrence of `Ti` replaced with
   from any of the `Bi`:
   - The least closure of `S` with respect to `L` is `U Function<X0 extends B0,
   ...., Xk extends Bk>(U0 x0, ...., Un1 xn, [Un+1 xn+1, ..., Um xm])` where:
-    - `U` is the least closure of `T` with respect to `L` 
-    - `Ui` is the greatest closure of `Ti` with respect to `L` 
+    - `U` is the least closure of `T` with respect to `L`
+    - `Ui` is the greatest closure of `Ti` with respect to `L`
     - with the usual capture avoiding requirement that the `Xi` do not appear in
   `L`.
   - The greatest closure of `S` with respect to `L` is `U Function<X0 extends B0,
   ...., Xk extends Bk>(U0 x0, ...., Un1 xn, [Un+1 xn+1, ..., Um xm])` where:
-    - `U` is the greatest closure of `T` with respect to `L` 
-    - `Ui` is the least closure of `Ti` with respect to `L` 
+    - `U` is the greatest closure of `T` with respect to `L`
+    - `Ui` is the least closure of `Ti` with respect to `L`
     - with the usual capture avoiding requirement that the `Xi` do not appear in
   `L`.
 - if `S` is `T Function<X0 extends B0, ...., Xk extends Bk>(T0 x0, ...., Tn xn,
-  {Tn+1 xn+1, ..., Tm xm})` and `L` does not contain any free type variables
+  {Tn+1 xn+1, ..., Tm xm})` and  
+  `L` does not contain any free type variables
   from any of the `Bi`:
   - The least closure of `S` with respect to `L` is `U Function<X0 extends B0,
   ...., Xk extends Bk>(U0 x0, ...., Un1 xn, {Un+1 xn+1, ..., Um xm})` where:
-    - `U` is the least closure of `T` with respect to `L` 
-    - `Ui` is the greatest closure of `Ti` with respect to `L` 
+    - `U` is the least closure of `T` with respect to `L`
+    - `Ui` is the greatest closure of `Ti` with respect to `L`
     - with the usual capture avoiding requirement that the `Xi` do not appear in
   `L`.
   - The greatest closure of `S` with respect to `L` is `U Function<X0 extends B0,
   ...., Xk extends Bk>(U0 x0, ...., Un1 xn, {Un+1 xn+1, ..., Um xm})` where:
-    - `U` is the greatest closure of `T` with respect to `L` 
-    - `Ui` is the least closure of `Ti` with respect to `L` 
+    - `U` is the greatest closure of `T` with respect to `L`
+    - `Ui` is the least closure of `Ti` with respect to `L`
     - with the usual capture avoiding requirement that the `Xi` do not appear in
   `L`.
+- if `S` is `T Function<X0 extends B0, ...., Xk extends Bk>(T0 x0, ...., Tn xn,
+    [Tn+1 xn+1, ..., Tm xm])` or `T Function<X0 extends B0, ...., Xk extends Bk>(T0 x0, ...., Tn xn,
+{Tn+1 xn+1, ..., Tm xm})`and `L` contains any free type variables
+  from any of the `Bi`:
+  - The least closure of `S` with respect to `L` is `Never`
+  - The greatest closure of `S` with respect to `L` is `Object?`
```

Comment:
I guess this could be `Function`.

Reply:
True, probably better.  cc @scheglov note the change.